### PR TITLE
Amazon: Implement UI Tests

### PR DESCRIPTION
**Pull request by moollaza**
## Description of new Instant Answer, or changes
Implements new UI Experiment by using a new custom Spice template that allows for a Brand link.

---
Instant Answer Page: https://duck.co/ia/view/products

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/amazon/branded_products_item.handlebars [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<div class="tile  tile--pr tile--c has-detail  tile--{{parentId}} opt--t-xxs" data-link="{{url}}">
```
Comment (moollaza) [inline share/spice/amazon/branded_products_item.handlebars]: For the purpose of the experiment, I've simply copied this from the open-source template to allow for easily adding the new Brand link when needed.

If the test is a success, the Products template should be updated to allow for the Brand link

**Comment by moollaza** [comment]
@bbraithwaite I've added style for the dark theme 👍 

juicer at duckduckgo 2017-12-06 20-54-12

File: share/spice/amazon/amazon.js [hunk 2]
```diff
@@ -16,8 +16,9 @@
         };
 
         if (api_result.results.length > 1) {
+            var isBrandExp = DDG.page.ads.adxExperiment === 'prod_brand_v1';
```
Comment (bbraithwaite) [inline share/spice/amazon/amazon.js]: We can use env here over DDG.

Also I'm pretty sure that env.page.ads will never cause a null reference error here, but it's worth double, double checking!

File: share/spice/amazon/branded_products_item.handlebars (hunk 1 above)
Comment (bbraithwaite) [reply share/spice/amazon/branded_products_item.handlebars]: 👍 Nice approach, much cleaner than adding IF then THIS etc to the other template.

File: share/spice/amazon/amazon.js (hunk 2 above)
Comment (moollaza) [reply share/spice/amazon/amazon.js]: > We can use env here over DDG.

@bbraithwaite I don't think so? AFAIK `env === window` here. So it'd need to be `env.DDG.ads`

>but it's worth double, double checking!

I'll add an extra check 👍

Comment (bbraithwaite) [reply share/spice/amazon/amazon.js]: @moollaza ah you're right re: `DDG` vs `env`, I'm so used to the internal code creating the IIFE with (DDG). 👍

**Comment by bbraithwaite** [comment]
👍

**Pull request merged**